### PR TITLE
[#161184357] Upgrade AWS Terraform provider

### PR DESCRIPTION
**Pull request by alext**
What
----

This upgrades the AWS Terraform provider and fixes a workaround that's no longer needed.

This is a precursor to switching to using Terraform to generate ACM certs. The version of the provider we're currently running doesn't support the necessary resources to enable this.

I've also taken the opportunity to DRY the pipeline image_resource definitions.

How to review
-------------

Run the pipeline from this branch, verify everything works.

## 🚨 Before merging 🚨 

* Merge https://github.com/alphagov/paas-docker-cloudfoundry-tools/pull/145
* Update the TMP commit.

Who can review
--------------

Not me.

### CODE REVIEW

File: concourse/pipelines/create-bosh-concourse.yml [hunk 1]
```diff
@@ -1,4 +1,47 @@
...
+    bosh-cli-v2: &gov-paas-bosh-cli-v2-image-resource
+      type: docker-image
+      source:
+        repository: governmentpaas/bosh-cli-v2
+        tag: 93e168b4244873902cd552ae08fc4a67a7b2e6f5
+    certstrap: &certstrap-image-resource
+      type: docker-image
+      source:
+        repository: governmentpaas/certstrap
+        tag: 895cf6752c8ec64af05a3a735186b90acd3db65a
```
Comment (bandesz) [inline concourse/pipelines/create-bosh-concourse.yml:18]: We had two different tags previously and 465642da06051a55630d39c899697b678f66a7f7 is the newer one, so I think we should use that.

**Review by bandesz — CHANGES_REQUESTED** [review]
We should amend the tag for the certstrap image

**Comment by bandesz** [comment]
There is one image reference in concourse-lite-self-terminate.yml. I think we should extract the image resource the same way, so it's consistent in all pipelines.

File: concourse/pipelines/create-bosh-concourse.yml (hunk 1 above)
Comment (alext) [reply concourse/pipelines/create-bosh-concourse.yml:18]: I picked the one that was used in more places. If you look at [the diff](https://github.com/alphagov/paas-docker-cloudfoundry-tools/compare/895cf67...465642d), you can see that there are actually no changes to the certstrap container between those versions.

**Comment by alext** [comment]
I've updated the remaining image_resource references...

**Review by bandesz — APPROVED** [review]

**Comment by bandesz** [comment]
I amended the tmp commit with the latest terraform image tag.

**Pull request merged**